<ui:composition
        xmlns="http://www.w3.org/1999/xhtml"
        xmlns:ui="http://xmlns.jcp.org/jsf/facelets"
        xmlns:h="http://xmlns.jcp.org/jsf/html"
        xmlns:c="http://xmlns.jcp.org/jsp/jstl/core"
        xmlns:b="http://bootsfaces.net/ui"
        template="/WEB-INF/jsf-templates/layout.xhtml">

    <ui:param name="title" value="Internal Server Error"/>

    <ui:define name="maincontent">
        <b:jumbotron>
            <h1>HTTP Error 500 - Internal Server Error</h1>
        </b:jumbotron>

        <h2>Oops!</h2>
        <p>
            Unfortunately, an unexpected problem has occurred during processing of your request.
            This is not your fault, <strong>this is our fault</strong>.
            We have been notified about this and we will try to investigate and fix it.
        </p>
        <p>
            You can try going back to <a href="#{requestScope['javax.servlet.error.request_uri']}">the same page</a>,
            or you can restart at <h:link outcome="/index">the home page</h:link>.
        </p>

        <c:if test="#{facesContext.application.projectStage eq 'Development'}">
            <h3>Error Detail for developer</h3>
            <dl>
                <dt>Request URI</dt>
                <dd>#{requestScope['javax.servlet.error.request_uri']}</dd>
                <dt>Exception type</dt>
                <dd>#{requestScope['javax.servlet.error.exception']['class']}</dd>
                <dt>Exception message</dt>
                <dd>#{requestScope['javax.servlet.error.exception'].message}</dd>
                <dt>Stack trace</dt>
                <dd><pre>#{
                        facesContext.externalContext.response.writer.flush()
                        }#{
                        requestScope['javax.servlet.error.exception'].printStackTrace
                                (facesContext.externalContext.response.writer)
                        }</pre>
                </dd>
            </dl>
        </c:if>


    </ui:define>

</ui:composition>
